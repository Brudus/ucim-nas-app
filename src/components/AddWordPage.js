import React from 'react';
import { connect } from 'react-redux';
import WordForm from './WordForm';
import { startAddWord } from '../actions/words';

export class AddWordPage extends React.Component {
    onSubmit = (word) => {
        this.props.startAddWord(word);
    };

    render() {
        return (
            <div>
                <h1>Add Word</h1>
                <WordForm 
                    onSubmit={this.onSubmit}
                />
            </div>
        );
    };
}

const mapDispatchToProps = (dispatch) => ({
    startAddWord: (word) => dispatch(startAddWord(word))
});

export default connect(undefined, mapDispatchToProps)(AddWordPage);