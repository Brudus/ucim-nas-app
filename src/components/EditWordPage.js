import React from 'react';
import { connect } from 'react-redux';
import WordForm from './WordForm';
import { startEditWord, startRemoveWord } from '../actions/words';

export class EditWordPage extends React.Component {
    onSubmit = (word) => {
        this.props.startEditWord(this.props.word.id, word);
        this.props.history.push('/catalog');
    };
    onRemove = (id) => {
        this.props.startRemoveWord({ id: this.props.word.id });
        this.props.history.push('/catalog');
    };

    render() {
        return (
            <div>
                <WordForm 
                    word={this.props.word}
                    onSubmit={this.onSubmit}
                />
                <button onClick={this.onRemove}>Remove</button>
            </div>
        );
    };
};

const mapStateToProps = (state, props) => ({
    word: state.words.find((word) => word.id === props.match.params.id)
});

const mapDispatchToProps = (dispatch, props) => ({
    startEditWord: (id, word) => dispatch(startEditWord(id, word)),
    startRemoveWord: (data) => dispatch(startRemoveWord(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditWordPage);