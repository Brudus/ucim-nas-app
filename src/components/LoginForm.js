import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export class LoginForm extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            email: '',
            password: '',
            loginError: '',
            emailPassed: undefined
        };
    }

    validateEmail = (email) => {
        const regex = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
        return regex.test(email);
    };

    onEmailChange = (e) => {
        const email = e.target.value;
        const emailPassed = this.validateEmail(email);
        this.setState(() => ({ 
            email,
            emailPassed
        }));
    };

    onPasswordChange = (e) => {
        const password = e.target.value;
        this.setState(() => ({ password }));
    }

    onSubmit = (e) => {
        e.preventDefault();

        this.setState(() => ({ 
            loginError: '',
            emailPassed: true
        }));
        this.props.startLogin(this.state.email, this.state.password);
    };

    render() {
        return (
            <div>
                <form onSubmit={this.onSubmit}>
                    {this.state.loginError && <p>{this.state.loginError}</p>}
                    <div>
                        <input 
                            type="email"
                            placeholder="Email address" 
                            autoFocus
                            onChange={this.onEmailChange}
                        />
                    </div>
                    
                    <div>
                        <input 
                            type="password"
                            placeholder="Password" 
                            onChange={this.onPasswordChange}
                        />
                    </div>
                    <button disabled={!this.state.emailPassed || !this.state.password}>Login</button>
                </form> 
            </div>
        );
    };
};

const mapDispatchToProps = (dispatch, props) => ({
    startLogin: (email, password) => dispatch(startLogin(email, password))
});

export default connect(undefined, mapDispatchToProps)(LoginForm);