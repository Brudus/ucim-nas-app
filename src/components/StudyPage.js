import React from 'react';
import { connect } from 'react-redux';

export const StudyPage = (props) => (
    <div>
        <h2>Study!</h2>
        {props.words[0].source}
    </div>
);

const mapStateToProps = (state) => ({
    words: state.words
});

export default connect(mapStateToProps)(StudyPage);