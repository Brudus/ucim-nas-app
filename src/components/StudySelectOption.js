import React from 'react';

export default class StudySelectionOption extends React.Component {
    handleClick = () => {
        this.props.calculateNewSchedule(this.props.grade);
    };

    render() {
        return (
            <div>
                <button onClick={this.handleClick}>{this.props.text}</button>
            </div>
        );
    };
};