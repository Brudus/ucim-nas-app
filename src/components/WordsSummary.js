import React from 'react';
import { connect } from 'react-redux';
import selectWords from '../selectors/words';
import filterWordsDue from '../selectors/words-due';

export const WordsSummary = ({ wordsCount, dueWordsCount }) => {
    const filteredWordsText = wordsCount === 1 ? '1 word' : `${wordsCount} words`;
    const filteredDueWordsText = dueWordsCount === 1 ? '1 word' : `${dueWordsCount} words`;

    return (
        <div>
            <h2>Viewing {filteredWordsText} with {filteredDueWordsText} due.</h2>
        </div>
    );
};

const mapStateToProps = (state) => {
    const visibleWords = selectWords(state.words, state.filters);

    return {
        wordsCount: visibleWords.length,
        dueWordsCount: filterWordsDue(visibleWords)
    };
};

export default connect(mapStateToProps)(WordsSummary);