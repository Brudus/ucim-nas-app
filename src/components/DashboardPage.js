import React from 'react';
import WordList from './WordList';
import WordListFilters from './WordListFilters';
import WordsSummary from './WordsSummary';

const DashboardPage = () => (
    <div>
        <WordsSummary />
        <WordListFilters />
        <WordList />
    </div>
);

export default DashboardPage;