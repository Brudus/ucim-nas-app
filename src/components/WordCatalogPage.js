import React from 'react';
import WordList from './WordList';
import WordListFilters from './WordListFilters';
import WordsSummary from './WordsSummary';

const WordCatalogPage = () => (
    <div>
        <WordsSummary />
        <WordListFilters />
        <WordList />
    </div>
);

export default WordCatalogPage;