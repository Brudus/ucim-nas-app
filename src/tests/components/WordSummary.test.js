import React from 'react';
import{ shallow } from 'enzyme';
import { WordsSummary } from '../../components/WordsSummary';

test('should render WordSummary correctly with 1 word', () => {
    const wrapper = shallow(<WordsSummary wordsCount={1} dueWordsCount={1} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render WordSummary correctly with multiple words', () => {
    const wrapper = shallow(<WordsSummary wordsCount={4} dueWordsCount={2} />);
    expect(wrapper).toMatchSnapshot();
});