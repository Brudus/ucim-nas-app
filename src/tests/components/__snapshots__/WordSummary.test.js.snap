// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render WordSummary correctly with 1 word 1`] = `
<div>
  <h2>
    Viewing 
    1 word
     with 
    1 word
     due.
  </h2>
</div>
`;

exports[`should render WordSummary correctly with multiple words 1`] = `
<div>
  <h2>
    Viewing 
    4 words
     with 
    2 words
     due.
  </h2>
</div>
`;
